cagent.go

```diff
@@ -59,6 +63,23 @@ func New(cfg *Config, version string) *Cagent {
 		}
 	}
 
+	if err := InitVMStat(); err != nil {
+		log.Error("cannot instantiate virtual machines API: ", err.Error())
+	}
+
+	for _, name := range ca.Config.VirtualMachinesStat {
+		vm, err := vmstat.Acquire(name)
+		if err != nil {
+			log.WithError(err).Warnf("acquire vm provider \"%s\"", name)
+		} else {
+			if err = vm.IsAvailable(); err != nil {
+				log.WithError(err).Warnf("vm provider \"%s\" either not available or not enabled for host system %s", name, runtime.GOOS)
```

Comment:
we don't use `logrus.WithError` in other parts of code. IMO this is not for user-facing errors and more suitable for central logging systems like logstash.

Instead, we describe the context of error first and then append the original error after semicolon

Reply:
good point, agree

Reply:
done